### PR TITLE
adding the method, docs to follow

### PR DESCRIPTION
**Pull request by console0**
Enables marketing calls.

## Deployment

on `irc.whapps.com`

```
!install catalogapi/perl/Whapps-CatalogAPI
```

### CODE REVIEW

File: perl/Whapps-CatalogAPI/lib/Whapps/CatalogAPI.pm [hunk 1]
```diff
@@ -142,6 +143,25 @@ sub catalog_breakdown
     return $ref;
 }
 
+=head2 catalog_marketing
+
+    my $catalog_marketing_response = $api->catalog_marketing(
+        socket_id => $socket_id,
+        page      => 'shop' );
+
+    TODO add its output/usage
```
Comment (zornme) [inline perl/Whapps-CatalogAPI/lib/Whapps/CatalogAPI.pm]: `TODO`

Comment (console0) [reply perl/Whapps-CatalogAPI/lib/Whapps/CatalogAPI.pm]: Done, actually did the docs.

**Comment by mccarrmb** [comment]
### Verification

Tests were conducted in a full end-to-end manner with the following code deployed simultaneously and tested on `mynrgrewards.online-rewards-qa.com`:

* https://github.com/Whapps/catalogapi/pull/3
* https://github.com/Whapps/incentive_base/pull/260
* https://github.com/Whapps/mynrgrewards/pull/72

#### Regression Tests

Default Landing


NRG Spouse Landing


Sub-category With No Marketing Options


#### Functional Tests

Category - Digital Cards:


Category - Electronics:


Category - Home & Garden:


Category - Sports & Outdoors:


Category - Toys & Games:


Category - Videogames:


**Category Marketing Functions**

Sub-category Search Option:


Sub-Category Link Option:


---

### Risks

* This code change will be deployed with other changes as detailed above. Said changes must be deployed in the following order or errors / support requests _could_ result:

1. https://github.com/Whapps/catalogapi/pull/3
2. https://github.com/Whapps/incentive_base/pull/260
3. https://github.com/Whapps/mynrgrewards/pull/72

---

### Result

QA :+1:

**Comment by Xavious** [comment]
Deployed

**Pull request merged**